src/perf_counter/perf_counter_atomic.h

```diff
@@ -214,12 +219,17 @@ class perf_counter_number_percentile_atomic : public perf_counter
             "pegasus_perf_counter_number_percentile_atomic counters");
         _timer.reset(new boost::asio::deadline_timer(tools::shared_io_service::instance().ios));
         _timer->expires_from_now(
-            boost::posix_time::seconds(rand() % _counter_computation_interval_seconds + 1));
+            boost::posix_time::seconds(::rand() % _counter_computation_interval_seconds + 1));
         _timer->async_wait(std::bind(
             &perf_counter_number_percentile_atomic::on_timer, this, _timer, std::placeholders::_1));
     }
 
-    ~perf_counter_number_percentile_atomic(void) { _timer->cancel(); }
+    ~perf_counter_number_percentile_atomic(void)
+    {
+        if (_timer) {
```

Comment:
for the if statement, it`s better using `if (_timer != nullptr)` if the type of _time is a pointer

Reply:
> for the if statement, it`s better using `if (_timer != nullptr)` if the type of _time is a pointer

Since `_timer` is a `std::shared_ptr`, we can use `operator bool `(https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool) to check if a std::shared_ptr has stored a valid pointer, shall we ?

Reply:
OK